litestar/connection/request.py

```diff
@@ -177,18 +185,23 @@ async def form(self) -> FormMultiDict:
             return FormMultiDict(self._form)
         content_type, options = self.content_type
         if content_type == RequestEncodingType.MULTI_PART:
-            self._form = self.scope["_form"] = form_values = parse_multipart_form(  # type: ignore[typeddict-unknown-key]
+            form_values = parse_multipart_form(
                 body=await self.body(),
                 boundary=options.get("boundary", "").encode(),
                 multipart_form_part_limit=self.app.multipart_form_part_limit,
             )
-            return FormMultiDict(form_values)
-        if content_type == RequestEncodingType.URL_ENCODED:
-            self._form = self.scope["_form"] = form_values = parse_url_encoded_form_data(  # type: ignore[typeddict-unknown-key]
+        elif content_type == RequestEncodingType.URL_ENCODED:
+            form_values = parse_url_encoded_form_data(
                 await self.body(),
             )
-            return FormMultiDict(form_values)
-        return FormMultiDict()
+        else:
+            form_values = {}
+
+        if form_values:
+            self._form = form_values
```

Comment:
```suggestion
        if self._form := form_values:
```
Would this be okay?
Muh walrus!